agent/consul/txn_endpoint.go

```diff
@@ -194,6 +194,14 @@ func (t *Txn) Read(args *structs.TxnReadRequest, reply *structs.TxnReadResponse)
 	if err != nil {
 		return err
 	}
+
+	// There are currently two different ways we handle permission issues.
+	//
+	// For simple reads such as KVGet and KVGetTree, the txn succeeds but the
+	// offending results are omitted. For more involved operations such as
+	// KVCheckIndex, the txn fails and permission denied errors are returned.
+	//
+	// TODO: Maybe we should unify these, or at least cover it in the docs?
```

Comment:
Please keep me honest here, is this right? Have I missed it in the docs? 😅

Reply:
hm, I couldn't find something obvious on https://www.consul.io/api-docs/txn

Was there another place you had checked?

Reply:
If it comes to it, let's throw an issue in our backlog regarding docs update/ refactor work.

Reply:
Nah, that's where I looked too 😄 agree let's pop it in the backlog (didn't want to tackle now because it's a breaking change to that API's contract).